src/OpenTelemetry.Contrib.Instrumentation.MassTransit/Implementation/MassTransitDiagnosticListener.cs

```diff
@@ -42,9 +46,33 @@ public override void OnStartActivity(Activity activity, object payload)
                 return;
             }
 
-            activity.DisplayName = this.GetDisplayName(activity);
+            string displayName = this.GetDisplayName(activity);
+            ActivityContext activityContext = default;
+            Activity originalActivityCurrent = Activity.Current;
```

Comment:
Should we use an activity instance from the method's parameters, instead of the global Activity?

Reply:
fixed